src/Type/Php/MinMaxFunctionReturnTypeExtension.php

```diff
@@ -51,7 +51,7 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 				if (!$isIterable->yes()) {
 					$argumentTypes[] = new ConstantBooleanType(false);
 				}
-				if ($iterableValueType instanceof UnionType) {
+				if ($iterableValueType instanceof UnionType && $argType instanceof ConstantArrayType) {
```

Comment:
This isn't gonna work if you have union of constant arrays. Please test that scenario + fix it with `TypeUtils::getOldConstantArrays()`.

Reply:
I wasn't sure how you wanted that I use getOldConstantArrays.

I handled UnionType in a different way and added a test. Tell me if this is ok this way.

Reply:
It's not OK, please do not use any `instanceof *Type`. Use `TypeUtils::getOldConstantArrays()` and iterate over the array of types.

Reply:
I tried a solution with getOldConstantArrays now.
But this seems more complicated to me.